File: Ejercicios/Java/Interfaces/Usuario.java
interface ISerializable {
    String serialize();
}

public class Usuario implements ISerializable {
    private String nombre;

    public Usuario() {
        this("Invitado");
    }

    public Usuario(String nombre) {
        this.nombre = nombre;
    }

    public Usuario(String nombre, int edad) {
        this.nombre = nombre + " (" + edad + ")";
    }

    public Usuario(Usuario otro) {
        this.nombre = otro.nombre;
    }

    @Override
    public String serialize() {
        return nombre;
    }
}

File: Ejercicios/GO/Clases/Usuario.go
package main

type Usuario struct {
	Nombre string
	Edad   int
}

func NewUsuario() Usuario {
	return Usuario{Nombre: "Invitado", Edad: 0}
}

func NewUsuarioConDatos(nombre string, edad int) Usuario {
	return Usuario{Nombre: nombre, Edad: edad}
}

func NewUsuarioSoloNombre(nombre string) Usuario {
	return Usuario{Nombre: nombre, Edad: 0}
}

func NewUsuarioCopy(u Usuario) Usuario {
	return Usuario{Nombre: u.Nombre, Edad: u.Edad}
}
